test/test_utility_functions.py

```diff
@@ -152,24 +152,17 @@ def test_does_not_increment_without_stream_send(self):
 
 class TestExtractHeader(object):
 
-    example_request_headers = [
-            (u':authority', u'example.com'),
-            (u':path', u'/'),
-            (u':scheme', u'https'),
-            (u':method', u'GET'),
-    ]
     example_headers_with_bytes = [
             (b':authority', b'example.com'),
             (b':path', b'/'),
             (b':scheme', b'https'),
             (b':method', b'GET'),
     ]
 
-    @pytest.mark.parametrize(
-        'headers', [example_request_headers, example_headers_with_bytes]
-    )
-    def test_extract_header_method(self, headers):
-        assert extract_method_header(headers) == b'GET'
+    def test_extract_header_method(self):
+        assert extract_method_header(
+            self.example_headers_with_bytes
+        ) == b'GET'
```

Comment:
applies to all changed tests: 

I think we are missing tests now for the case where headers get changed using the newly introduced `utf8_encode_headers` function. Please add relevant tests to ensure string headers always get correctly encoded into bytes.

Reply:
Since I made all utility functions operate on bytes, I don't think we are missing any coverage here: they don't work with unicode and we don't have unicode tests. If we had typing, we could have enforced the "bytes-only" headers thing but we don't have them (yet?). Coverage report also says 100%:

```
---------- coverage: platform linux, python 3.10.12-final-0 ----------
Name                                                         Stmts   Miss Branch BrPart  Cover   Missing
--------------------------------------------------------------------------------------------------------
.tox/py310/lib/python3.10/site-packages/h2/__init__.py           1      0      0      0   100%
.tox/py310/lib/python3.10/site-packages/h2/config.py            54      0      8      0   100%
.tox/py310/lib/python3.10/site-packages/h2/connection.py       622      0    152      0   100%
.tox/py310/lib/python3.10/site-packages/h2/errors.py            22      0      0      0   100%
.tox/py310/lib/python3.10/site-packages/h2/events.py           137      0      4      0   100%
.tox/py310/lib/python3.10/site-packages/h2/exceptions.py        44      0      0      0   100%
.tox/py310/lib/python3.10/site-packages/h2/frame_buffer.py      62      0     20      0   100%
.tox/py310/lib/python3.10/site-packages/h2/settings.py         131      0     38      0   100%
.tox/py310/lib/python3.10/site-packages/h2/stream.py           447      0     88      0   100%
.tox/py310/lib/python3.10/site-packages/h2/utilities.py        225      0    130      0   100%
.tox/py310/lib/python3.10/site-packages/h2/windows.py           35      0     12      0   100%
--------------------------------------------------------------------------------------------------------
TOTAL                                                         1780      0    452      0   100%
Coverage XML written to file coverage.xml

Required test coverage of 100.0% reached. Total coverage: 100.00%
```

That said I'll do another pass to make sure we have normal `str` inputs to headers from upstream functions. I definitely saw a few examples but I'll make sure we cover both cases in tests for higher up functions.

Reply:
I originally typed "coverage" into my comment but then thought about it some more - and I agree it is likely we have 100% coverage - but since library users might pass in `str` headers, do we miss some cases somehow? While the encoding code path seems to be taken at least once - I'm not sure if we check, assert, and verify on all important code paths.

I need to go back through everything and check our public API definition and if everything is still as expected in terms of accepted & expected types.

Reply:
Okay, went through the entire test suite and added bytes and str headers to all tests utilizing any request headers.

Reply:
>I need to go back through everything and check our public API definition and if everything is still as expected in terms of accepted & expected types.

I think I've saved you the hassle as I went through all tests and made sure we used both `str` and `bytes` inputs from all high-level usage 😄